From ec03013a086eb2ac3b882a10aaf59579b12b24f8 Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Thu, 28 Jan 2021 11:16:39 +0200
Subject: [PATCH 1226/1921] arm64: marvell_v8_sdk_defconfig: bypass SMMU by
 default

Make sure the SMMU bypass is enabled by default, so the DMA masters
that are not configured to use IOMMU, continue to work as usual.

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I86935b595f65ecdad121ee8e2dacd811b065af3e
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/44914
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/configs/marvell_v8_sdk_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/configs/marvell_v8_sdk_defconfig b/arch/arm64/configs/marvell_v8_sdk_defconfig
index dcb51ef354e8..edf3eb811ce2 100644
--- a/arch/arm64/configs/marvell_v8_sdk_defconfig
+++ b/arch/arm64/configs/marvell_v8_sdk_defconfig
@@ -712,6 +712,7 @@ CONFIG_QCOM_APCS_IPC=y
 CONFIG_ROCKCHIP_IOMMU=y
 CONFIG_TEGRA_IOMMU_SMMU=y
 CONFIG_ARM_SMMU=y
+# CONFIG_ARM_SMMU_DISABLE_BYPASS_BY_DEFAULT is not set
 CONFIG_ARM_SMMU_V3=y
 CONFIG_QCOM_IOMMU=y
 CONFIG_REMOTEPROC=y
-- 
2.31.1

